### PR TITLE
Issue3139 flushing

### LINKED ISSUE #3139: Running flusher job fails with panic
```
level=error ts=2020-09-04T17:58:54.070116286Z caller=flush.go:241 org_id=0 msg="failed to flush user" err="failed to write chunk, 1 values remaining: context deadline exceeded"
level=error ts=2020-09-04T17:58:58.154123166Z caller=flush.go:241 org_id=0 msg="failed to flush user" err="failed to write chunk, 1 values remaining: context deadline exceeded"
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x10c9182]
goroutine 684 [running]:
github.com/cortexproject/cortex/pkg/util/validation.(*Overrides).getOverridesForUser(...)
/go/src/github.com/cortexproject/cortex/pkg/util/validation/limits.go:355
github.com/cortexproject/cortex/pkg/util/validation.(*Overrides).MinChunkLength(0x0, 0x5b4e3d0, 0x1, 0x0)
/go/src/github.com/cortexproject/cortex/pkg/util/validation/limits.go:321 +0x22
github.com/cortexproject/cortex/pkg/ingester.(*Ingester).flushUserSeries(0xc000589800, 0x26, 0x5b4e3d0, 0x1, 0x42d1484788b2d9b8, 0x0, 0xc01ac55e00, 0x0, 0x0)
/go/src/github.com/cortexproject/cortex/pkg/ingester/flush.go:294 +0x2b0
github.com/cortexproject/cortex/pkg/ingester.(*Ingester).flushLoop(0xc000589800, 0x26)
/go/src/github.com/cortexproject/cortex/pkg/ingester/flush.go:238 +0x106
created by github.com/cortexproject/cortex/pkg/ingester.(*Ingester).startFlushLoops
/go/src/github.com/cortexproject/cortex/pkg/ingester/ingester.go:257 +0x68
```

(reported by user)

### PR DESCRIPTION
**Pull request by pstibrany**
**What this PR does**: This PR fixes panic in flusher component. When flusher initializes ingester component in "for flusher" mode, all ingester loops still run... especially `FlushCheckPeriod`. This check may find that chunks are idle (no new samples appended for some time), and put them onto the queue. If flushing of  "immediate" chunk fails, it it put back to queue -- but now it's behind its own "idle" version. For idle chunks there is additional check that requires limits. Flusher however doesn't set any limits, so this failed.

This PR modifies ingester to avoid running some loops (flush-check, metadata-purge and rate-update loop) when running in "for flusher" mode. That way ingester still doesn't need limits, and no panic occurs because queue will not contain "idle" chunks. Just in case, we also set limits, but unit test passes `nil` limits still.

**Which issue(s) this PR fixes**:
Fixes #3139 

**Checklist**
- [x] Tests updated
- [ ] Documentation added
- [ ] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: pkg/ingester/ingester.go [hunk 1]
```diff
@@ -273,9 +273,10 @@ func NewForFlusher(cfg Config, chunkStore ChunkStore, registerer prometheus.Regi
 		chunkStore:  chunkStore,
 		flushQueues: make([]*util.PriorityQueue, cfg.ConcurrentFlushes),
 		wal:         &noopWAL{},
+		limits:      limits,
```
Comment (codesome) [inline pkg/ingester/ingester.go:276]: Why do we need limits for flusher?

Comment (pstibrany) [reply pkg/ingester/ingester.go:276]: Missing limits is what caused panic in the referenced issue. This PR is fixing that (limits were used by "Idle" series, this PR stops the loop that can generate them), but it also sets limits to avoid panics if flusher code accidentally tries to use them again.

**Review by pracucci — APPROVED** [review]
LGTM, thanks!
